pkg/pod/container.go

```diff
@@ -280,6 +280,25 @@ func (builder *ContainerBuilder) WithResourceRequest(hugePages, memory string, c
 	return builder
 }
 
+// WithCustomResourcesLimits applies custom resource limit struct on container.
+func (builder *ContainerBuilder) WithCustomResourcesLimits(resourceList v1.ResourceList) *ContainerBuilder {
+	glog.V(100).Infof("Applying custom resource limit to container: %v", resourceList)
+
+	if resourceList == nil {
+		glog.V(100).Infof("Container's resource limit var 'resourceList' is empty")
+
+		builder.errorMsg = "container's resource limit var 'resourceList' is empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.definition.Resources.Limits = resourceList
```

Comment:
not related to this PR, but the ContainerBuilder struct has `definition` instead of `Definition`

Reply:
Yes, I did it intentionally. container.builder is private struct that it not exposed

Reply:
Thanks for clarification